cache/contenthash/checksum.go

```diff
@@ -941,7 +941,7 @@ func (cc *cacheContext) checksum(ctx context.Context, root *iradix.Node, txn *ir
 		}
 
 		// no FollowSymlinkInScope because invalid paths should not be inserted
-		fp := filepath.Join(target, filepath.FromSlash(p))
+		fp := path.Join(target, filepath.FromSlash(p))
```

Comment:
Rest of the changes smgt but this one I don't understand. This is the actual FS path for syscalls and `filepath.FromSlash` has already created a backslash path.

Reply:
You are correct. I got too eager with replacing ```filepath.Join()```. Reverted this bit.